Automattic/woocommerce-payments
# Hide WooPay button on zero total orders without shipping address

Fixes #7895

#### Changes proposed in this Pull Request








This PR hides the WooPay button on zero total orders, or when buying zero total subscription (0 sign up fee and 0 renewal price) when the cart does not need shipping.

This PR also fixes an issue where other express payment buttons do not get hidden when the order gets to 0 after applying gift cards on classic checkout.

#### Testing instructions





* Create a $0 virtual product.
* Access this product page.
* The WooPay express checkout button should be hidden.
* Add this product to the cart.
* Access the cart or checkout pages on blocks and classic checkout.
* The WooPay express checkout button should be hidden.
* Interact with WooPay email field on checkout page, it should not check if the email exists or redirect if you are already authenticated. 
* Add another virtual product with any value greater than 0.
* Access the cart or checkout pages on blocks and classic checkout.
* The WooPay express checkout button should be shown.
* Add coupons and gift cards to make the order total decrease to 0.
* The WooPay express checkout button should be hidden (using gift cards on classic checkout now should hide the other express payment buttons too).
* Create a $0 virtual subscription, without sign up fee.
* Access this subscription page.
* The WooPay express checkout button should be hidden.
* Add this subscription to the cart.
* Access the cart or checkout pages on blocks and classic checkout.
* The WooPay express checkout button should be hidden.
* Interact with WooPay email field on checkout page, it should not check if the email exists or redirect if you are already authenticated.
* Change the sign up fee to any value greater than 0.
* Reload the cart or checkout page.
* The WooPay express checkout button should be shown.
* For any other scenario, the WooPay express checkout button should work as it worked before.

-------------------


**Post merge**
